app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
**OPTIONAL**
- It'd be nice if you created a custom method using the `after_save` rails callback to call the `update_post_counter` method every time the model gets updated!

Ex:
`after_save :update_post_counter`

> Please consider this suggestion in the other models